src/components/ui/modelControlPanel.tsx

```diff
@@ -0,0 +1,65 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import * as React from "react";
+import { Button } from "~/components/ui/button";
+import { useEffect, useState } from "react";
+import { cn } from "~/lib/utils";
+
+export interface ModelControlPanelProps {
+  stages: { number: number; name: string }[];
+}
+
+const ModelControlPanel = React.forwardRef<
+  HTMLDivElement,
+  ModelControlPanelProps
+>(({ stages }, ref) => {
+  const [currentStageNumber, setCurrentStageNumber] = useState(1);
+  const [isAreaView, setIsAreaView] = useState(true);
+  const router = useRouter();
+
+  useEffect(() => {
+    router.push(
+      `/refa/stages/${currentStageNumber}?view=${isAreaView ? "areas" : "artefacts"}`,
+    );
+  }, [currentStageNumber, isAreaView, router]);
+
+  const handleToggle = () => {
+    setIsAreaView(!isAreaView);
+  };
+
+  const handleStageClick = (stageNumber: number) => {
+    setCurrentStageNumber(stageNumber);
+  };
+
+  return (
+    <div className="flex items-center justify-between">
+      <div className="-ml-3 -mr-40 flex min-w-full">
+        {stages.map((stage) => (
+          <div key={stage.name} className="border-gray-200 py-4">
+            <div className="flex flex-row">
+              <span className="inline-flex items-center">
+                <Button
+                  variant={"link"}
+                  onClick={() => handleStageClick(stage.number)}
+                  className={cn(
+                    "text-md font-medium",
+                    currentStageNumber === stage.number ? "underline" : "",
+                  )}
+                >
+                  {stage.name}
+                </Button>
```

Comment:
Why are we using a button for changing the stage instead of the previous logic that used `<Link />`? I believe the previous implementation was already quite robust.

I feel that we can make it less complicated if we setup Zustand (which would take 2 steps) and just keep a global store for the artefact/area view instead of tracking it by URL. It will also allow Nextjs to better manage caching of server components. (if we can avoid using useEffect, it's for the best)

Reply:
Added Zustand as you suggested